### PR TITLE
ReadMe - Adding table of contents 

### PR DESCRIPTION
**Pull request by nikithauc**
Adding a table of contents and some minor updates to the ReadMe.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,60 +81,25 @@ In case your browser doesn't have support for [Fetch](https://developer.mozilla.
...
-For more information on initializing client, refer [this document](./docs/CreatingClientInstance.md).
 
-### 4. Make requests to the graph
+### 3. Make requests to the graph
 
-Once you have authentication setup and an instance of Client, you can begin to make calls to the service. All requests should be start with `client.api(path)` and end with an [action](./docs/Actions.md).
+Once you have authentication setup and an instance of Client, you can begin to make calls to the service. All requests should start with `client.api(path)` and end with an [action](./docs/Actions.md).
 
-Getting user details
+Getting user details -
```
Comment (zengin) [inline README.md:102]: Q: what is the trailing ` -` sign for?

File: README.md [hunk 2]
```diff
@@ -109,7 +110,7 @@ try {
 }
 ```
 
-Sending an email to the recipients
+Sending an email to the recipients -
```
Comment (zengin) [inline README.md:113]: same here

File: docs/CreatingClientInstance.md [hunk 3]
```diff
@@ -42,11 +50,14 @@ let clientOptions: ClientOptions = {
...
-Refer, [custom authentication provider](./CustomAuthenticationProvider.md) for more detailed information.
-
 ### Option B. Custom Middleware chain
 
-Want to have complete control over the request and the response objects, one can provide his own chain of middleware. Have to pass first middleware in the chain as `middleware` in [ClientOptions](../src/IClientOptions.ts).
+The Microsoft Graph SDK client allows configuring custom middleware, allowing you to change the behavior of the client. For example, you can insert customized logging, or add a test handler to simulate specific scenarios.
+
+To create a client instance with the custom middleware chain -
+
+1. Refer, [custom middleware chain](./CustomMiddlewareChain.md) for more detailed information.
```
Comment (zengin) [inline docs/CreatingClientInstance.md:59]: nit:
```suggestion
1. Refer to [custom middleware chain](./CustomMiddlewareChain.md) for more detailed information.
```

File: docs/CreatingClientInstance.md [hunk 4]
```diff
@@ -4,35 +4,43 @@ Initialization of the Client can be done in one of below two ways
...
 ## 1. Create With ClientOptions [Recommended]
 
+The Microsoft Graph SDK client configures a default set of middleware that allows the SDK to communicate with the Microsoft Graph endpoints. This default set is customizable, allowing you to change the behavior of the client
+
 In order to instantiate a Client object, one has to pass in the `authProvider` or `middleware` chain in [ClientOptions](../src/IClientOptions.ts).
 
 ### Option A. Default Middleware chain
 
-Pass an instance of a class which implements [AuthenticationProvider](../src/IAuthenticationProvider.ts) interface as `authProvider` in [ClientOptions](../src/IClientOptions.ts), which will instantiate the Client with default set of middleware chain.
+The default middleware chain contains consecutively chained instances of the following -
```
Comment (zengin) [inline docs/CreatingClientInstance.md:13]:
```suggestion
The default middleware chain contains consecutively chained instances of the following:
```

File: docs/CreatingClientInstance.md [hunk 5]
```diff
@@ -4,35 +4,43 @@ Initialization of the Client can be done in one of below two ways
...
+- [AuthenticationHandler](../src/middleware/AuthenticationHandler.ts)
+- [RetryHandler](../src/middleware/RetryHandler.ts)
+- [RedirectHandler](../src/middleware/RedirectHandler.ts)
+- [TelemetryHandler](../src/middleware/TelemetryHandler.ts)
+- [HTTPMessageHandler](../src/middleware/HTTPMessageHandler.ts)
 
-Library is shipped with one such authentication provider named [ImplicitMSALAuthenticationProvider](../src/ImplicitMSALAuthenticationProvider.ts). This ImplicitMSALAuthenticationProvider depends on an authentication library [msal.js](https://github.com/AzureAD/microsoft-authentication-library-for-js) which is not shipped along with the library, one has to externally include msal.js to use ImplicitMSALAuthenticationProvider.
+To create a client instance with the default middleware chain -
```
Comment (zengin) [inline docs/CreatingClientInstance.md:20]:
```suggestion
To create a client instance with the default middleware chain:
```

File: docs/CreatingClientInstance.md [hunk 6]
```diff
@@ -42,11 +50,14 @@ let clientOptions: ClientOptions = {
...
 ```
 
-Refer, [custom authentication provider](./CustomAuthenticationProvider.md) for more detailed information.
-
 ### Option B. Custom Middleware chain
 
-Want to have complete control over the request and the response objects, one can provide his own chain of middleware. Have to pass first middleware in the chain as `middleware` in [ClientOptions](../src/IClientOptions.ts).
+The Microsoft Graph SDK client allows configuring custom middleware, allowing you to change the behavior of the client. For example, you can insert customized logging, or add a test handler to simulate specific scenarios.
+
+To create a client instance with the custom middleware chain -
```
Comment (zengin) [inline docs/CreatingClientInstance.md:57]:
```suggestion
To create a client instance with the custom middleware chain:
```

**Review by zengin — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (nikithauc) [reply README.md:102]: Indicating the the following sample is for Getting user details

**Pull request merged**